pyquil/experiment/_main.py

```diff
@@ -168,63 +182,65 @@ def __init__(
         else:
             self.reset = False
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._settings)
 
-    def __getitem__(self, item):
+    def __getitem__(self, item: int) -> List[ExperimentSetting]:
         return self._settings[item]
 
-    def __setitem__(self, key, value):
+    def __setitem__(self, key: int, value: List[ExperimentSetting]) -> None:
         self._settings[key] = value
 
-    def __delitem__(self, key):
+    def __delitem__(self, key: int) -> None:
         self._settings.__delitem__(key)
 
-    def __iter__(self):
+    def __iter__(self) -> Generator[List[ExperimentSetting], None, None]:
         yield from self._settings
 
-    def __reversed__(self):
+    def __reversed__(self) -> Generator[List[ExperimentSetting], None, None]:
         yield from reversed(self._settings)
 
-    def __contains__(self, item):
+    def __contains__(self, item: List[ExperimentSetting]) -> bool:
         return item in self._settings
 
-    def append(self, expts):
+    def append(self, expts: Union[ExperimentSetting, List[ExperimentSetting]]) -> None:
         if not isinstance(expts, list):
             expts = [expts]
         return self._settings.append(expts)
 
-    def count(self, expt):
+    def count(self, expt: List[ExperimentSetting]) -> int:
```

Comment:
If you ever want to subclass `ExperimentSetting`, these `List[ExperimentSetting]` input arg types could become painful. On the other hand, so is changing to `Sequence` and then having to copy them everywhere.

Reply:
problem for the future

Reply:
this can wait til 2020 no'm sayin?